boards/arm/gd32f407v_start/doc/index.rst

```diff
@@ -0,0 +1,142 @@
+.. _gd32f407v_start:
+
+GigaDevice GD32F407V-START
+##########################
+
+Overview
+********
+
+The GD32F407V-START board is a hardware platform that enables prototyping
+on GD32F407VE Cortex-M4 High Performance MCU.
+
+The GD32F407VE features a single-core ARM Cortex-M4 MCU which can run up
+to 168 MHz with flash accesses zero wait states, 3072kiB of Flash, 192kiB of
+SRAM and 80 GPIOs.
+
+.. image:: img/gd32f407v_start.jpg
+     :align: center
+     :alt: gd32f407v_start
+
+Hardware
+********
+
+- GD32F407VET6 MCU
+- 1 x User LEDs
+- 1 x User Push buttons
+- GD-Link on board programmer
+- J-Link/SWD connector
+
+For more information about the GD32F407 SoC and GD32F407V-START board:
+
+- `GigaDevice Cortex-M4 High Performance SoC Website`_
+- `GD32F407X Datasheet`_
+- `GD32F40X User Manual`_
+- `GD32F407V-START User Manual`_
+
+Supported Features
+==================
+
+The board configuration supports the following hardware features:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Peripheral
+     - Kconfig option
+     - Devicetree compatible
+   * - EXTI
+     - :kconfig:option:`CONFIG_GD32_EXTI`
+     - :dtcompatible:`gd,gd32-exti`
+   * - GPIO
+     - :kconfig:option:`CONFIG_GPIO`
+     - :dtcompatible:`gd,gd32-gpio`
+   * - NVIC
+     - N/A
+     - :dtcompatible:`arm,v7m-nvic`
+   * - PWM
+     - :kconfig:option:`CONFIG_PWM`
+     - :dtcompatible:`gd,gd32-pwm`
+   * - SYSTICK
+     - N/A
+     - N/A
+   * - USART
+     - :kconfig:option:`CONFIG_SERIAL`
+     - :dtcompatible:`gd,gd32-usart`
+
+Other peripherals may be used if shields are connected to the board.
+
+Serial Port
+===========
+
+The GD32F407V-START board has one serial communication port. The default port
+is USART0 with TX connected at PB6 and RX at PB7.
+
+Programming and Debugging
+*************************
+
+Before programming your board make sure to configure boot jumpers as
+follows:
+
+- JP3/4: Select 2-3 for both (boot from user memory)
+
+Using J-Link
+=============
+
+While the board comes with an embedded GD-Link programmer, it is not yet
+supported by OpenOCD. However, J-Link can be used to program the board using
+the SWD interface via PA13(SWDIO) and PA14(SWCLK) exposed in the JP6 header.
```

Comment:
isn't it supported on f407 really?

Reply:
I understand it as "openocd not OFFICIALLY support GD32".
https://sourceforge.net/p/openocd/mailman/message/37687417/

I think using pyOCD (#49594) is preferable.
So, I reworked this PR referencing #49594.